src/Storages/ExternalStream/Kafka/KafkaSink.cpp

```diff
@@ -304,8 +287,8 @@ void KafkaSink::addMessageToBatch(char * pos, size_t len)
         .partition = next_partition,
         .payload = const_cast<void *>(static_cast<const void *>(payload)),
         .len = len,
-        .key = const_cast<void *>(static_cast<const void *>(key.data())),
-        .key_len = key.size(),
+        .key = const_cast<void *>(static_cast<const void *>(key.data)),
```

Comment:
do we need all of this cast ?

Reply:
Same for payload

Reply:
The difference here is that `key.data` has type of `const char *`, if we remove the `const_cast`, it will report `Cannot initialize a member subobject of type 'void *' with an lvalue of type 'const char *'. But the `static_cast` can be removed.